src/rules/declaration-property-value-no-unknown/index.js

```diff
@@ -0,0 +1,287 @@
+const isCustomProperty = require("../../utils/isCustomPropertySet.js");
+const isStandardSyntaxDeclaration = require("../../utils/isStandardSyntaxDeclaration.js");
+const isStandardSyntaxProperty = require("../../utils/isStandardSyntaxProperty.js");
+const isStandardSyntaxValue = require("../../utils/isStandardSyntaxValue.js");
+const matchesStringOrRegExp = require("../../utils/matchesStringOrRegExp.js");
+const atKeywords = require("../../utils/atKeywords.js");
+const validateObjectWithArrayProps = require("../../utils/validateObjectWithArrayProps.js");
+const { utils } = require("stylelint");
+const {
+  isDollarVar,
+  isIfStatement,
+  isNestedProperty
+} = require("../../utils/validateTypes");
+const namespace = require("../../utils/namespace");
+const ruleUrl = require("../../utils/ruleUrl");
+
+const ruleName = namespace("declaration-property-value-no-unknown");
+
+const messages = utils.ruleMessages(ruleName, {
+  rejected: (property, value) =>
+    `Unexpected unknown value "${value}" for property "${property}"`,
+  rejectedParseError: (property, value) =>
+    `Cannot parse property value "${value}" for property "${property}"`
+});
+
+const meta = {
+  url: ruleUrl(ruleName)
+};
+
+const SYNTAX_PROPERTY = /^syntax$/i;
+
+function extractFunctionName(inputString) {
+  const matches = [...inputString.matchAll(/(?:\s*([\w\-$]+)\s*)?\(/g)].flat();
+  return matches;
+}
+
+const unsupportedFunctions = ["clamp", "min", "max", "env"];
+
+function rule(primary, secondaryOptions) {
+  return (root, result) => {
+    const validOptions = utils.validateOptions(
+      result,
+      ruleName,
+      { actual: primary },
+      {
+        actual: secondaryOptions,
+        possible: {
+          ignoreProperties: [validateObjectWithArrayProps],
+          propertiesSyntax: [isPlainObject.isPlainObject],
+          typesSyntax: [isPlainObject.isPlainObject]
+        },
+        optional: true
+      }
+    );
+
+    if (!validOptions) {
+      return;
+    }
+
+    const ignoreProperties = Array.from(
+      Object.entries(secondaryOptions?.ignoreProperties ?? {})
+    );
+
+    /** @type {(name: string, propValue: string) => boolean} */
+    const isPropIgnored = (name, value) => {
+      const [, valuePattern] =
+        ignoreProperties.find(([namePattern]) =>
+          matchesStringOrRegExp(name, namePattern)
+        ) || [];
+
+      return valuePattern && matchesStringOrRegExp(value, valuePattern);
+    };
+
+    const propertiesSyntax = {
+      // Take a shallow clone as this object will be appended to.
+      ...(secondaryOptions?.propertiesSyntax ?? {})
+    };
+    const typesSyntax = secondaryOptions?.typesSyntax ?? {};
+
+    /** @type {Map<string, string>} */
+    const typedCustomPropertyNames = new Map();
+
+    // Unless we tracked return values of declared functions, they're all valid.
+    root.walkAtRules("function", atRule => {
+      unsupportedFunctions.push(extractFunctionName(atRule.params)[1]);
```

Comment:
Hmm... To make it easier to keep the rule up-to-date with the original rule, it would be a bit cleaner to keep the unsupported function check from the stylelint rule, and instead push/map these to a separate array.

Reply:
Do you mean export the function directly from Stylelint? 🤔 
This is how the function looked originally
```
function containsUnsupportedFunction(cssTreeNode) {
	return Boolean(
		cssTree.find(
			cssTreeNode,
			(node) => node.type === 'Function' && ['clamp', 'min', 'max', 'env'].includes(node.name),
		),
	);
}
```
I extracted that array into a global array and I push the function names to that same array.

Reply:
Sorry for the confusing explanation, let me try again:

Since we are copying the rule from Stylelint's repo, it would be good to keep the original rule as untouched as possible and then make the additional changes on top of that. That way it would be easier to maintain the rule in case the original rule changes.

So what I'm suggesting is that the `containsUnsupportedFunction` would remain the same as in the original rule, you would push the function names to an array, and add another early return next to `if (containsUnsupportedFunction(cssTreeValueNode)) return;`

Reply:
Ohh okay gotcha, makes sense. Thanks for explaining further :)